### DISCUSSION TITLE
How do I always select SDH English subs if found but fall back to regular English if SDH is not found?

### DISCUSSION THREAD

**Discussion by Pentaphon** [CogentRedTester/mpv-sub-select#15] [Q&A] [answered]
@CogentRedTester or anybody else out there, I have a deaf friend who prefers SDH English subs but if he can't get them, would like to always fall back to regular English subs. How would he config this?

English SDH subs are hard-of-hearing subs like "Text # 2" in the following mediainfo:

Let's say a file only has the English subs but not SDH. If you whitelist to always select SDH, that works, but if SDH is not there, subs are disabled instead of having the regular English sub selected. Can conditions help me? 

```
Text #1
ID                                       : 3
Format                                   : UTF-8
Codec ID                                 : S_TEXT/UTF8
Codec ID/Info                            : UTF-8 Plain Text
Duration                                 : 1 h 29 min
Bit rate                                 : 30 b/s
Frame rate                               : 0.141 FPS
Count of elements                        : 754
Stream size                              : 19.7 KiB (0%)
Language                                 : English
Default                                  : No
Forced                                   : No
```

```
Text #2
ID                                       : 4
Format                                   : UTF-8
Codec ID                                 : S_TEXT/UTF8
Codec ID/Info                            : UTF-8 Plain Text
Duration                                 : 1 h 36 min
Bit rate                                 : 34 b/s
Frame rate                               : 0.168 FPS
Count of elements                        : 968
Stream size                              : 24.4 KiB (0%)
Title                                    : SDH
Language                                 : English
Default                                  : No
Forced                                   : No
```

**Comment by CogentRedTester** [comment, marked as answer]
Just add a second english entry to the json file without a whitelist.

**Reply by Pentaphon** [reply]
> Just add a second english entry to the json file without a whitelist.

~~Do you mean another set of brackets like this under it without the whitelist? I just tried that on the file with the earlier mediainfo and it didn't work. The following results in sub track being disabled. Not sure what I'm doing wrong.~~

**Edit: this was my mistake, I've posted the correct, working json for my use case below.** 

```
[
    {
        "alang": "*",
        "slang": "eng?",
        "whitelist": [ "sdh" ]
    },
    {
        "alang": "*",
        "slang": "eng?"
    }
]
```

**Reply by CogentRedTester** [reply]
That should have worked, yes. Can you upload a log file generated with `--log-file=log.txt --msg-level=sub_select='trace'`.

**Reply by CogentRedTester** [reply]
As you said in what I assume was a deleted comment you had invalid JSON syntax.

**Reply by CogentRedTester** [reply]
So is it fixed like you said in the deleted comment?

**Reply by Pentaphon** [reply]
> So is it fixed like you said in the deleted comment?

No I double checked just to be sure and it wasn't fixed (so I deleted the comment) in the file you see in the log so please look at it, sorry.

**Reply by CogentRedTester** [reply]
In the log file you posted the script crashed due to invalid JSON formatting. If you're using the exact JSON that you posted above then I'd suggest you add the missing comma after the first `"slang": "eng?"` and remove the trailing comma after the second `"slang": "eng?"`.

**Reply by Pentaphon** [reply]
> In the log file you posted the script crashed due to invalid JSON formatting. If you're using the exact JSON that you posted above then I'd suggest you add the missing comma after the first `"slang": "eng?"` and remove the trailing comma after the second `"slang": "eng?"`.

Ah, so I did actually post the wrong syntax here. Thanks for the correction. It works now. Thanks for your help. I'll delete my log now for privacy reasons but I'll post the updated json so other people can copy the correct syntax if they have my same issue.
